src/mqt/bench/devices/rigetti.py

```diff
@@ -128,36 +135,69 @@ def import_backend(cls, path: Path) -> Device:
         device.name = rigetti_calibration["name"]
         device.num_qubits = rigetti_calibration["num_qubits"]
         device.basis_gates = rigetti_calibration["basis_gates"]
+
+        is_M3 = "m3" in device.name
         device.coupling_map = [
-            [cls.__from_rigetti_index(a), cls.__from_rigetti_index(b)] for a, b in rigetti_calibration["connectivity"]
+            [cls.__from_rigetti_index(a, is_M3), cls.__from_rigetti_index(b, is_M3)]
+            for a, b in rigetti_calibration["connectivity"]
         ]
 
         calibration = DeviceCalibration()
         for qubit in range(device.num_qubits):
-            rigetti_index = cls.__to_rigetti_index(qubit)
+            rigetti_index = cls.__to_rigetti_index(qubit, is_M3)
             calibration.single_qubit_gate_fidelity[qubit] = {
                 gate: rigetti_calibration["properties"]["1Q"][str(rigetti_index)]["f1QRB"] for gate in ["rx", "rz"]
             }
             calibration.readout_fidelity[qubit] = rigetti_calibration["properties"]["1Q"][str(rigetti_index)]["fRO"]
             calibration.t1[qubit] = rigetti_calibration["properties"]["1Q"][str(rigetti_index)]["T1"]
             calibration.t2[qubit] = rigetti_calibration["properties"]["1Q"][str(rigetti_index)]["T2"]
 
+        # To calculate the average value for missing two-qubit gate fidelities
+        cz_lst, cp_lst, xx_plus_yy_lst = [], [], []
+        msg = (
+            "Rigetti device fidelity data is not available for some two-qubit gates."
+            "The average value of the available gates will be used for the missing ones."
+        )
+        warnings.warn(msg, stacklevel=1)
+
         for qubit1, qubit2 in device.coupling_map:
-            rigetti_index1 = cls.__to_rigetti_index(qubit1)
-            rigetti_index2 = cls.__to_rigetti_index(qubit2)
+            rigetti_index1 = cls.__to_rigetti_index(qubit1, is_M3)
+            rigetti_index2 = cls.__to_rigetti_index(qubit2, is_M3)
             if qubit1 > qubit2:
                 continue
 
             edge = f"{rigetti_index1}-{rigetti_index2}"
             fidelity = {}
-            with suppress(KeyError):
+            try:
                 fidelity["cz"] = rigetti_calibration["properties"]["2Q"][edge]["fCZ"]
-            with suppress(KeyError):
+                cz_lst.append(fidelity["cz"])
```

Comment:
I am not sure I understand why those `_lst` variables are created. Could you briefly explain it to me?

Reply:
They make it easy to claculate the average values for the missing ones.
Instead we could store numbers instead and use the update rule like `fid_new += fid_before + ...`.
I think the lists are a bit cleaner bc we would need to introduce a counter for each gate type.